pkgs/misc/emulators/ryujinx/default.nix

```diff
@@ -0,0 +1,108 @@
+{ stdenv, fetchFromGitHub , fetchurl , makeWrapper , makeDesktopItem , linkFarmFromDrvs
+, dotnet-sdk_3 , dotnetPackages , dotnetCorePackages
+, SDL2 , libX11 , openal
+, gtk3 , gobject-introspection , wrapGAppsHook
+}:
+let
+  runtimeDeps = [
+    SDL2
+    gtk3
+    libX11
+    openal
+  ];
+in
+stdenv.mkDerivation rec {
+  pname = "ryujinx";
+  version = "1.0.5094";
+
+  src = fetchFromGitHub {
+    owner = "Ryujinx";
+    repo = "Ryujinx";
+    rev = "16bab8fb886673608d950a765f07a706cb506103";
+    sha256 = "0a9a8dz752f7rmklyiirar68llz48wcawm5gl8r5g5dal4q28j6m";
+  };
+
+  nativeBuildInputs = [ dotnet-sdk_3 dotnetPackages.Nuget makeWrapper wrapGAppsHook gobject-introspection ];
+
+  nugetDeps = linkFarmFromDrvs "${pname}-nuget-deps" (import ./deps.nix {
+    fetchNuGet = { name, version, sha256 }: fetchurl {
+      name = "nuget-${name}-${version}.nupkg";
+      url = "https://www.nuget.org/api/v2/package/${name}/${version}";
+      inherit sha256;
+    };
+  }
+  );
+
+  patches = [ ./log.patch ]; # Without this, Ryujinx tries to write logs to the nix store. This patch makes it write to the directory Ryujinx is launched from.
```

Comment:
Is there anyway to get them to log to a normal directory on the platform? E.g. XDG on linux, whatever is used on windows?

Reply:
Done :+1: 

On Linux, it now writes logs to `~/.config/Ryujinx/Logs/`, and on other platforms it should use the right directory as well. This directory is already used for stuff like config files.

I'm not very familiar with C#, so I've based this off of [this](https://github.com/Ryujinx/Ryujinx/blob/60db4c353099e8656a330ede03fdbe57a421fa47/Ryujinx.HLE/FileSystem/VirtualFileSystem.cs#L214).